aten/src/ATen/native/mps/operations/LossOps.mm

```diff
@@ -771,12 +771,12 @@ void smooth_l1_loss_impl(
           MPSGraphTensor *mpsGraphOneTensor = [mpsGraph constantWithScalar: 1.0
                                                                   dataType: inputTensor.dataType];
           MPSGraphTensor *mpsGraphHalfTensor = [mpsGraph constantWithScalar: 0.5
-                                                                   dataType: MPSDataTypeFloat32];
+                                                                   dataType: inputTensor.dataType];
```

Comment:
Will integer data types work with this change?  (e.g if input was torch.int32)?

Reply:
Good point, but I think the loss ops only take float inputs - even in test consistency, the values associated with the key "nn.functional.smooth_l1_loss" are only torch.float16 and torch.float32.